src/Compilers/CSharp/Portable/Symbols/Source/SourceDelegateClonedParameterSymbolForBeginAndEndInvoke.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SourceDelegateClonedParameterSymbolForBeginAndEndInvoke : SourceClonedParameterSymbol
+    {
+        internal SourceDelegateClonedParameterSymbolForBeginAndEndInvoke(SourceParameterSymbol originalParam, SourceDelegateMethodSymbol newOwner, int newOrdinal)
+            : base(originalParam, newOwner, newOrdinal, suppressOptional: true)
+        {
+        }
+
+        internal override bool IsCallerFilePath => _originalParam.IsCallerFilePath;
+
+        internal override bool IsCallerLineNumber => _originalParam.IsCallerLineNumber;
+
+        internal override bool IsCallerMemberName => _originalParam.IsCallerMemberName;
+
+        // We don't currently support caller argument expression for cloned begin/end invoke.
```

Comment:
This is a "what" comment, not a "why" comment. We should include the information about how Begin/EndInvoke throw PNS exceptions and that we therefore don't do the index remapping for this case as we don't feel it's necessary.

Reply:
```suggestion
        // We don't currently support caller argument expression for cloned begin/end invoke since
        // they throw PlatformNotSupportedException at runtime and we feel it's unnecessary to support them.
```